src/layer_shims.cpp

```diff
@@ -583,13 +668,40 @@ XRAPI_ATTR XrResult XRAPI_CALL thisLayer_xrGetActionStateBoolean(XrSession sessi
 	return res;
 }
 
+void recordApplyHapticFeedback(XrHapticActionInfo* hapticActionInfo, XrHapticBaseHeader* hapticBaseHeader)
```

Comment:
I don't get the haptic feedback recording. When replaying a trace, the application should trigger haptic feedback also if none was recorded in the trace. I.e., the haptic feedback is a response to user input, not user input by itself.

Why do we want to include this in the trace?

Reply:
I explain th reason for this in the future work section of the lab report. tldr: if we want to sync the trace to the game we might want to replay untill we get to a haptic feedback and wait on a haptic feedback from the game as a rudimentary way of checking if we are actually performing the actions in the sequence that the game expects it. This might not be generalizable but i'm confident that it could work for beat saber. I can remove it from the pull request if you would prefer that. I does not get replayed.

Reply:
This is great! Can you please:

1. Add an experiment in your lab report where you use the haptic feedback events to validate the timing of librnr's playback? I would be very interested to see this.
2. Add an option that suppresses playback of haptic events (because we the application to also trigger a haptic event), which defaults to `true`?